curriculum/challenges/english/10-coding-interview-prep/project-euler/problem-128-hexagonal-tile-differences.md

```diff
@@ -28,27 +28,79 @@ Find the 2000th tile in this sequence.
 
 # --hints--
 
-`hexagonalTile()` should return `14516824220`.
+`hexagonalTile(10)` should return `271`.
 
 ```js
-assert.strictEqual(hexagonalTile(), 14516824220);
+assert.strictEqual(hexagonalTile(10), 271);
+```
+
+`hexagonalTile(2000)` should return `14516824220`.
+
+```js
+assert.strictEqual(hexagonalTile(2000), 14516824220);
 ```
 
 # --seed--
 
 ## --seed-contents--
 
 ```js
-function hexagonalTile() {
+function hexagonalTile(tileIndex) {
 
   return true;
 }
 
-hexagonalTile();
+hexagonalTile(10);
 ```
 
 # --solutions--
 
 ```js
-// solution required
+const NUM_PRIMES = 840000;
+const PRIME_SEIVE = Array(Math.floor((NUM_PRIMES-1)/2)).fill(true);
+(function initPrimes(num) {
+  const upper = Math.floor((num - 1) / 2);
+  const sqrtUpper = Math.floor((Math.sqrt(num) - 1) / 2);
+  for (let i = 0; i <= sqrtUpper; i++) {
+    if (PRIME_SEIVE[i]) {
+      // Mark value in PRIMES array
+      const prime = 2 * i + 3;
+      // Mark all multiples of this number as false (not prime)
+      const primeSqaredIndex = 2 * i ** 2 + 6 * i + 3;
+      for (let j = primeSqaredIndex; j < upper; j += prime) {
+        PRIME_SEIVE[j] = false;
+      }
+    }
+  }
+})(NUM_PRIMES);
+
+function isPrime(num) {
+  if (num === 2) return true;
+  else if (num % 2 === 0) return false
+  else return PRIME_SEIVE[(num - 3) / 2];
+}
```

Comment:
For the ones that actually need a prime seive and are a bit slow, we can swap in this prime sieve that uses skips the multiples of two